### PR TITLE
Replace ProposalObject with Chef::DataBag

### PR DESCRIPTION
**Pull request by bkutil**
The code uses ProposalObject api just to store the dispersion results in
a data bag. Use Chef::DataBag directly.

Refs https://github.com/crowbar/barclamp-crowbar/pull/1269

### CODE REVIEW

**Comment by vuntz** [comment]
+1

File: crowbar_framework/app/models/swift_service.rb [hunk 1]
```diff
@@ -306,7 +305,7 @@ def _get_proposal_by_node(node)
 
   def validate_proposal_after_save proposal
     # first, check for conflict with ceph
-    ProposalObject.find_proposals("ceph").each {|p|
+    Proposal.where(barclamp: "ceph").all.each {|p|
```
Comment (vuntz) [inline crowbar_framework/app/models/swift_service.rb]: Can we remove the `all` from here? ;-)

Comment (bkutil) [inline crowbar_framework/app/models/swift_service.rb]: Done.

Comment (vuntz) [inline crowbar_framework/app/models/swift_service.rb]: Interestingly, you removed it from another place, but not this one :-)

Comment (bkutil) [inline crowbar_framework/app/models/swift_service.rb]: > Interestingly, you removed it from another place, but not this one :-)

lol. hopefully fixed now :)

**Pull request merged**